tests/integration/security/reachability_test.go

```diff
@@ -89,6 +89,26 @@ func TestReachability(t *testing.T) {
 					ExpectMTLS:             Never,
 					SkippedForMulticluster: true,
 				},
+				{
+					ConfigFile: "beta-mtls-automtls-workload.yaml",
+					Namespace:  apps.Namespace1,
+					Include: func(src echo.Instance, opts echo.CallOptions) bool {
+						return (apps.D.Contains(src) || apps.IsNaked(src)) &&
+							(apps.A.Contains(opts.Target) || apps.B.Contains(opts.Target) || apps.C.Contains(opts.Target))
+					},
+					ExpectSuccess: func(src echo.Instance, opts echo.CallOptions) bool {
+						// Sidecar injected client always succeed.
+						if apps.D.Contains(src) {
+							return true
+						}
+						// For naked app as client, only requests targeted to mTLS disabled endpoints succeed:
+						// A are disabled by workload selector for entire service.
+						// C port 8090 http port are disabled.
+						return apps.A.Contains(opts.Target) || (apps.C.Contains(opts.Target) && opts.PortName == "http")
+					},
+					ExpectMTLS:             Never,
```

Comment:
should the `ExpectMTLS` be used for the mTLS cases?

Reply:
done. use mtlsOnExpect.